File: api/src/main/java/net/blueshell/api/listener/calendar/AddEventToCalendarEventListener.java
package net.blueshell.api.listener.calendar;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import net.blueshell.api.common.event.job.AddEventToCalendarEvent;
import net.blueshell.api.job.calendar.AddEventToCalendarJob;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

@Slf4j
@Component
@RequiredArgsConstructor
public class AddEventToCalendarEventListener {

    private final AddEventToCalendarJob job;

    @EventListener
    public void onAdd(AddEventToCalendarEvent evt) {
        Long id = evt.eventId();
        if (id == null) return;
        job.add(id);
    }
}

File: api/src/main/kotlin/net/blueshell/api/listener/calendar/AddEventToCalendarEventListener.kt
package net.blueshell.api.listener.calendar

import net.blueshell.api.common.event.job.AddEventToCalendarEvent
import net.blueshell.api.job.calendar.AddEventToCalendarJob
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class AddEventToCalendarEventListener(
    private val job: AddEventToCalendarJob
) {
    @EventListener
    fun onAdd(evt: AddEventToCalendarEvent) {
        val id = evt.eventId ?: return
        job.add(id)
    }
}
